### PR TITLE
Adds filter query parameter on dashboard widget for ObjectCountWidget

### LINKED ISSUE #12599: Carry filters over when clicking on object list in dashboard widget
### NetBox version

v3.5.1

### Feature type

Change to existing functionality

### Proposed functionality

I have an "Object Count" widget with a Devices/Cables/Racks/Sites and tenant filter `"{"tenant_id": "19"}`. I'd like if I click (for example) on "Devices" that I get `/dcim/devices/?tenant_id=19` instead of `/dcim/devices/`.

### Use case

Makes widgets with filters more useful if clicking on something gives a page with the actual items in the count instead of _all_ objects of that type. And can make the dashboard widgets a handy way to get to what that user needs to see.

### Database changes

_No response_

### External dependencies

_No response_

### PR DESCRIPTION
**Pull request by abhi1693**

### CODE REVIEW

**Comment by kkthxbye-code** [comment]
I'm not sure about this one. I get the intention, but the ObjectCountWidget filters are queryset filters. Blindly converting these to URL FilterSet filters seems like a bad idea. It will work in some cases, but there's a lot where it wont.

Example (IPAM > Prefixes):

{
    {"tenant__name": "Dunder-Mifflin, Inc."}
}

Resulting URL:

`http://127.0.0.1:8000/ipam/prefixes/?tenant__name=Dunder-Mifflin%2C+Inc.`

`tenant__name` is of course not an actual FilterSet filter on PrefixFilterSet so it does nothing.

I'm not sure there is a clean solution here, QuerySet filters and FilterSet filters will never be compatible, so we would need to switch to using the API, however there's no "count" API endpoints to my knowledge.

**Comment by abhi1693** [comment]
Good catch, I had completely forgotten that this runs different. The other solution (not preferable) is to split at `__` and query the object ID to transform the query params into IDs. That may have bad performance overall though.

**Comment by jeremystretch** [comment]
I've opened #12742 to address this, which IMO should be treated as a bug as it can be very confusing to end users.

**Comment by abhi1693** [comment]
Closing in favour of the new approach discussed here.

**Comment by jeremystretch** [comment]
Well, I think this PR is still needed, and will suffice once we've solved for #12742.

File: netbox/extras/dashboard/widgets.py [hunk 1]
```diff
@@ -180,8 +181,14 @@ def render(self, request):
             else:
                 counts.append((model, None))
 
+        filter_query_string = ''
+        if filters := self.config.get('filters', None):
+            filter_items = [(k, v) if not isinstance(v, list) else zip([k] * len(v), v) for k, v in filters.items()]
+            filter_query_string = urlencode(filter_items, doseq=True)
```
Comment (jeremystretch) [inline netbox/extras/dashboard/widgets.py:187]: We can leverage Django's QueryDict for this. Something like:

```
qd = QueryDict(mutable=True)
qd.update(filters)
filter_query_string = qd.urlencode()
```

**Pull request closed without merging**